docs/site/Implementing-auto-migration.md

```diff
@@ -0,0 +1,158 @@
+---
+lang: en
+title: 'Implementing Auto-Migration'
+keywords: LoopBack 4.0
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Implementing-auto-migration.html
+---
+
+## Overview
+
+In LoopBack, auto-migration helps the user create relational database schemas
+based on definitions of their models. Auto-migration can facilitate the
+synchronization of the database and models so that they match, such as in cases
+where the database needs to be changed in order to match the models. LoopBack
+offers two ways to do this:
+
+- **Auto-migrate**: Drop existing schema objects if they exist and re-create
+  them based on model definitions. Existing data will be lost.
+
+- **Auto-update**: Change database schema objects if there is a difference
+  between the objects and model definitions. Existing data will be kept.
+
+## Implementation Example
+
+There are two ways to implement
+[automigrate()](http://apidocs.loopback.io/loopback-datasource-juggler/#datasource-prototype-automigrate)
+and
+[autoupdate()](http://apidocs.loopback.io/loopback-datasource-juggler/#datasource-prototype-autoupdate),
+shown below with the
+[TodoList](https://loopback.io/doc/en/lb4/todo-list-tutorial.html) example.
+
+### First Approach: index.ts
+
+This approach is if you want to have migration centralized in one place.
+
+In **src/index.ts**, add the following import statement and import your models:
+
+```ts
+import {DataSource, Repository} from '@loopback/repository';
+import {Todo, TodoList} from './models';
+```
+
+Then get the datasource and your repositories:
+
+```ts
+const ds = await app.get<DataSource>('datasources.db');
+const todoRepo = await app.get<Repository<Todo>>('repositories.TodoRepository');
+const todoListRepo = await app.get<Repository<TodoList>>(
+  'repositories.TodoListRepository',
+);
+```
+
+Finally, call **automigrate()** or **autoupdate()**:
+
+```ts
+ds.automigrate((err: any) => {
```

Comment:
I believe that `automigrate` function returns a promise, thus it can be awaited. Let's avoid callbacks and stay consistent with the async/await flow control we use everywhere else.

```ts
await ds.automigrate();
```